Add a client class

djmetzle opened the pull request:
Add a client class.

djmetzle committed:
Add example client

Found here:
http://danknox.github.io/2013/01/27/using-rubys-native-nethttp-library/

This is a simple example client for a `net/http` only API wrapper.

We should further extened this to be more specific to the Replicated
Vendor API and it's semantics.

djmetzle committed:
Add token use to client

Add a method `set_token` to allow using a replicated API token. If this
token is set, use it in the `Authorization` header for the request.

Note: Don't pass this into constructor

djmetzle committed:
Merge remote-tracking branch 'origin/master' into implement-client

MooreGuy commented on lib/client/client.rb:
```diff
@@ -0,0 +1,54 @@
+      case method
+      when :get
+         full_path = encode_path_params(path, params)
+         request = VERB_MAP[method.to_sym].new(full_path)
```
string to lower so that "GET" or "get" will work?

g1y commented on lib/client/client.rb:
```diff
@@ -0,0 +1,54 @@
+      response
+   end
+
+   def request(method, path, params)
```
method to lower so it "GET" or "get" would work?

djmetzle committed:
Downcase HTTP verb

It looks cleaner to all caps HTTP verbs. So convert those caps strings
to lower, then to_sym in the request method.

MooreGuy commented:
CR :+1:

djmetzle commented:
Merging.
